sources/lib/structures/ty.mli

```diff
@@ -54,6 +54,9 @@ type t =
       to values of type [dst]. *)
   | Tsum of Hstring.t * Hstring.t list
   (** Enumeration, with its name, and the list of its constructors. *)
+
+  | Tadt of Hstring.t * t list
```

Comment:
Could you add a little doc comment describing the semantic and/or use case of this new constructor ?